part/txn.go

```diff
@@ -217,7 +217,7 @@ func (txn *Txn[T]) modify(root *header[T], key []byte, mod func(T) T) (oldValue
 				if this.watch != nil {
 					txn.watches[this.watch] = struct{}{}
 				}
-				this = this.promote(!txn.opts.rootOnlyWatch || this == newRoot)
+				this = this.promote(!txn.opts.rootOnlyWatch || this == root)
```

Comment:
not of consequence as far as I can tell since it's written to before read, but `thisp` in line 198 should probably point to `&root` instead of `&newRoot`? Or just nil, if it really is always written to before being read

Reply:
No it must point to `&newRoot` so that we'll update it correctly as we recurse into the old tree and clone it.

Reply:
oh duh, yeah I get it now :+1: